lighthouse-core/test/audits/uses-rel-preload-test.js

```diff
@@ -9,191 +9,206 @@
-    const scriptSubNode = new NetworkNode(networkRecords[4]);
-    const scriptOtherNode = new NetworkNode(networkRecords[5]);
-
-    mainDocumentNode.setIsMainDocument(true);
-    mainDocumentNode.addDependency(rootNode);
-    scriptNode.addDependency(mainDocumentNode);
-    scriptAddedNode.addDependency(scriptNode);
-    scriptSubNode.addDependency(scriptNode);
-    scriptOtherNode.addDependency(scriptNode);
-
-    mockGraph = rootNode;
-    mockSimulator = {
-      simulate(graph) {
-        const nodesByUrl = new Map();
-        graph.traverse(node => nodesByUrl.set(node.record.url, node));
-
-        const rootNodeLocal = nodesByUrl.get(rootNode.record.url);
-        const mainDocumentNodeLocal = nodesByUrl.get(mainDocumentNode.record.url);
-        const scriptNodeLocal = nodesByUrl.get(scriptNode.record.url);
-        const scriptAddedNodeLocal = nodesByUrl.get(scriptAddedNode.record.url);
-        const scriptSubNodeLocal = nodesByUrl.get(scriptSubNode.record.url);
-        const scriptOtherNodeLocal = nodesByUrl.get(scriptOtherNode.record.url);
-
-        const nodeTimings = new Map([
-          [rootNodeLocal, {starTime: 0, endTime: 500}],
-          [mainDocumentNodeLocal, {startTime: 500, endTime: 1000}],
-          [scriptNodeLocal, {startTime: 1000, endTime: 2000}],
-          [scriptAddedNodeLocal, {startTime: 2000, endTime: 3250}],
-          [scriptSubNodeLocal, {startTime: 2000, endTime: 3000}],
-          [scriptOtherNodeLocal, {startTime: 2000, endTime: 3500}],
-        ]);
-
-        if (scriptAddedNodeLocal.getDependencies()[0] === mainDocumentNodeLocal) {
-          nodeTimings.set(scriptAddedNodeLocal, {startTime: 1000, endTime: 2000});
-        }
-
-        if (scriptSubNodeLocal.getDependencies()[0] === mainDocumentNodeLocal) {
-          nodeTimings.set(scriptSubNodeLocal, {startTime: 1000, endTime: 2000});
-        }
-
-        if (scriptOtherNodeLocal.getDependencies()[0] === mainDocumentNodeLocal) {
-          nodeTimings.set(scriptOtherNodeLocal, {startTime: 1000, endTime: 2500});
-        }
-
-        return {timeInMs: 3500, nodeTimings};
-      },
-    };
-
-    return UsesRelPreload.audit(mockArtifacts(networkRecords, mainResource), {}).then(
+    const artifacts = mockArtifacts(networkRecords, mainDocumentNodeUrl);
+    return UsesRelPreload.audit(artifacts, {settings: {}}).then(
       output => {
-        assert.equal(output.rawValue, 1250);
+        assert.equal(output.details.overallSavingsMs, 330);
         assert.equal(output.details.items.length, 2);
-        assert.equal(output.details.items[0].url, 'http://www.example.com/script-added.js');
-        assert.equal(output.details.items[1].url, 'http://sub.example.com/script-sub.js');
+        assert.equal(output.details.items[0].url, scriptSubNodeUrl);
+        assert.equal(output.details.items[0].wastedMs, 330);
+        assert.equal(output.details.items[1].url, scriptAddedNodeUrl);
+        assert.equal(output.details.items[1].wastedMs, 180);
```

Comment:
@patrickhulce it seemed fine to change these to the timings derived from the generated network records, but I wasn't sure what the `if (scriptOtherNodeLocal.getDependencies()[0] === mainDocumentNodeLocal) {` conditionals were up to, so wanted to double check

Reply:
that was just so we'd only be setting the original timings and letting the altered simulation ones be inferred on there own.


this is much nicer :) 👍